s3/replication/common/experiments/kafka_consumer.py

```diff
@@ -0,0 +1,152 @@
+#
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# Kafka consumer code
+# Please note : Few changes are required in the replication
+# manager module to add this consumer code.
+# For more details, please refer the feature branch code:
+# https://github.com/priyanka25081999/cortx-multisite/tree/ps/kafka
+
+import json
+from ast import literal_eval
+from s3replicationmanager.prepare_job import PrepareReplicationJob
+# Make sure to install aiokafka module using the command
+# >> pip3 install aiokafka
+# aiokafka version used for testing : aiokafka==0.7.2
+from aiokafka import AIOKafkaConsumer
+from aiokafka.structs import OffsetAndMetadata, TopicPartition
+
+
+class KafkaMain:
+    consumer = None
+
+    '''
+    Job id to offset mapping
+      {
+        "job_id": { <topic> : <topic_name>,
+                    <partition> : <partition_name>
+                    <offset> : <offset_value>
+                  }
+            ...
+       }
+    '''
+    job_id_to_offset = {}
+    last_commited_offset = 0
+
+    def __init__(self):
+        """Initialize."""
+
+    # Initialize the kafka consumer
+    # This method should be called from app.py class on startup.
+    def initialize(self, app):
+        self._app = app
+        KafkaMain.consumer = AIOKafkaConsumer(
+            'FDMI',
+            bootstrap_servers='127.0.0.1:9092',
+            group_id="test",
+            auto_offset_reset='latest',
+            enable_auto_commit=False)
+
+    # Getter methods
+    def get_topic(self, job_id):
```

Comment:
What is this Codacy warning?
Do you have plan to fix it?

Reply:
Yes, I am checking the codacy warning - http://pylint-messages.wikidot.com/messages:r0201